client/my-sites/checkout/checkout-thank-you/pending.tsx

```diff
@@ -144,41 +185,136 @@ function useRedirectOnTransactionSuccess( {
 			return;
 		}
 
-		if ( redirectInstructions.isError ) {
-			const defaultFailErrorNotice = translate(
-				"Sorry, we couldn't process your payment. Please try again later."
-			);
-			reduxDispatch(
-				errorNotice( defaultFailErrorNotice, {
-					isPersistent: true,
-				} )
-			);
-		}
-
-		if ( redirectInstructions.isUnknown ) {
-			const unknownNotice = translate( 'Oops! Something went wrong. Please try again later.' );
-			reduxDispatch(
-				errorNotice( unknownNotice, {
-					isPersistent: true,
-				} )
-			);
-		}
-
 		didRedirect.current = true;
+		triggerPostRedirectNotices( {
+			redirectInstructions,
+			isRenewal,
+			productName,
+			willAutoRenew,
+			translate,
+			reduxDispatch,
+		} );
 		performRedirect( redirectInstructions.url );
 	}, [
 		error,
+		finalReceiptId,
+		isReceiptLoaded,
+		isRenewal,
+		orderId,
+		productName,
+		receiptId,
 		redirectTo,
 		reduxDispatch,
+		reloadCart,
 		siteSlug,
 		transaction,
 		translate,
-		reloadCart,
-		orderId,
-		receiptId,
+		willAutoRenew,
 	] );
 }
 
+function isTransactionSuccessful(
+	transaction: OrderTransaction | null
+): transaction is OrderTransactionSuccess {
+	return transaction?.processingStatus === SUCCESS;
+}
+
+function triggerPostRedirectNotices( {
+	redirectInstructions,
+	isRenewal,
+	productName,
+	willAutoRenew,
+	translate,
+	reduxDispatch,
+}: {
+	redirectInstructions: RedirectInstructions;
+	isRenewal: boolean;
+	productName: string;
+	willAutoRenew: boolean;
+	translate: ReturnType< typeof useTranslate >;
+	reduxDispatch: CalypsoDispatch;
+} ): void {
+	if ( redirectInstructions.isError ) {
+		const defaultFailErrorNotice = translate(
+			"Sorry, we couldn't process your payment. Please try again later."
+		);
+		reduxDispatch(
+			errorNotice( defaultFailErrorNotice, {
+				displayOnNextPage: true,
+			} )
+		);
+		return;
+	}
+
+	if ( redirectInstructions.isUnknown ) {
+		const unknownNotice = translate( 'Oops! Something went wrong. Please try again later.' );
+		reduxDispatch(
+			errorNotice( unknownNotice, {
+				displayOnNextPage: true,
+			} )
+		);
+		return;
+	}
+
+	if ( isRenewal ) {
+		displayRenewalSuccessNotice( {
+			productName,
+			willAutoRenew,
+			translate,
+			reduxDispatch,
+		} );
+		return;
+	}
+
+	reduxDispatch(
+		successNotice( translate( 'Your purchase has been completed!' ), {
+			displayOnNextPage: true,
+		} )
+	);
+}
+
+function displayRenewalSuccessNotice( {
+	productName,
+	willAutoRenew,
+	translate,
+	reduxDispatch,
+}: {
+	productName: string;
+	willAutoRenew: boolean;
+	translate: ReturnType< typeof useTranslate >;
+	reduxDispatch: CalypsoDispatch;
+} ): void {
+	if ( willAutoRenew ) {
+		// showing notice for product that will auto-renew
+		reduxDispatch(
+			successNotice(
+				translate( 'Success! You renewed %(productName)s. {{a}}Learn more about renewals{{/a}}', {
```

Comment:
Previously this read `'Success! You renewed %(productName)s for %(duration)s, and we sent your receipt to %(email)s. {{a}}Learn more about renewals{{/a}}'`.

I've shortened these renewal success messages because the receipt does not have access to the billing term, expiry date, or email address, but I don't think those were really necessary and I think the benefit outweighs the loss (redirect methods were getting no notice at all before). We can always add that info back in later if we find a way to access it on this page.